gemini/gemini_browser.py

```diff
@@ -232,14 +242,45 @@ def db_schema():
     return template('db_schema.j2')
 
 
-def browser_main(parser, args):
-    global database
+## Switch between the different available browsers
+def browser_puzzle(args):
+    host = args.host
+    port = args.port
 
-    print "!!!!!"
-    print "NOTE: open a browser and point it to http://localhost:8088/query"
-    print "!!!!!"
+    plugin = GeminiPlugin(db=args.db, vtype="sv")
+    root = os.path.expanduser("~/.puzzle")
 
-    database = args.db
+    BaseConfig.PUZZLE_BACKEND = plugin
+    BaseConfig.UPLOAD_DIR = os.path.join(root, 'resources')
+
+    puzzle_srv = puzzle_app.create_app(config_obj=BaseConfig)
+    webbrowser.open_new_tab("http://{}:{}".format(host, port))
+    run(puzzle_srv, host=host, port=port)
+
+def browser_builtin(args):
+    host = args.host
+    port = args.port
 
-    run(app, host='localhost', port=8088,
+    webbrowser.open_new_tab("http://{}:{}".format(host, port))
+    run(app, host=host, port=port,
         reloader=True, debug=True)
+
+
+def browser_main(parser, args):
+    global database
+    database = args.db
+    browser = args.use
+
+    try:
+        if args.use == "puzzle":
+            browser_puzzle(args)
+        # XXX: https://github.com/dgaston/kvasir
```

Comment:
Thanks, I'll take a look into it. Mine is actually a fairly large application on its own that interfaces just through the GEMINI API. I'll have to dust it off and integrate over the next few weeks